### PR TITLE
Make conditional logic dropdown options label/value update with field options

### PR DESCRIPTION
**Pull request by AbdiTolesa**
Fix https://github.com/Strategy11/formidable-pro/issues/2911
Requires https://github.com/Strategy11/formidable-pro/pull/4940

### Test steps

1. Create a new form with a radio button field. Check 'Use separate values'.
2. Add a conditional field based off of the radio.
3. Update the the radio options to something else, including the option used in the conditional logic in (2).
4. The conditional logic value options should be updated to reflect the change.
5. Confirm that the field labels and values are mapped to the conditional logic dropdown correctly

### CODE REVIEW

**Review by garretlaxton — APPROVED** [review]
This works great! The conditional logic dropdowns show the updated field values.

File: js/formidable_admin.js [hunk 1]
```diff
@@ -5501,18 +5547,30 @@ function frmAdminBuildJS() {
...
 				}
 
-				valueSelect.prepend( optionMatch );
+				if ( optionMatch ) {
+					valueSelect.prepend( optionMatch );
+				}
 			}
 
 			optionMatch = valueSelect.querySelector( 'option[value=""]' );
+			console.log('Empty option', optionMatch);
```
Comment (truongwp) [inline js/formidable_admin.js:5573]: There is a debug code here.

File: js/formidable_admin.js [hunk 2]
```diff
@@ -5480,6 +5506,26 @@ function frmAdminBuildJS() {
...
+	 * @returns {Object}
+	 */
+	const getNewConditionalLogicOption = ( fieldId, expectedOption ) => {
+		const optionsContainer = document.getElementById( 'frm_field_' + fieldId + '_opts' );
+
+		const expectedOptionContainer = optionsContainer.querySelector( 'input[value="' + expectedOption + '"]' );
+
+		if ( expectedOptionContainer ) {
+			const choiceComponents = getChoiceValueAndLabel( expectedOptionContainer );
+			return { value: choiceComponents.value, label: choiceComponents.label };
```
Comment (truongwp) [inline js/formidable_admin.js:5523]: Can we just return `choiceComponents` here?

**Comment by truongwp** [comment]
@AbdiTolesa I got 2 problems:

1. When adding a new option, I change the label, but it isn't updated in the conditional logic.
2. When deleting an option, it isn't removed in the conditional logic.

Please see my screen recording:

https://github.com/Strategy11/formidable-forms/assets/19748318/2846ff7c-4383-4d3f-8bca-b9a577bf1478

**Comment by AbdiTolesa** [comment]
> @AbdiTolesa I got 2 problems:
> 
> 1. When adding a new option, I change the label, but it isn't updated in the conditional logic.
> 2. When deleting an option, it isn't removed in the conditional logic.

@truongwp Thanks for catching those issues! They are both fixed now if you can check again.

File: js/formidable_admin.js [hunk 3]
```diff
@@ -3520,6 +3520,28 @@ function frmAdminBuildJS() {
...
+	 *
+	 * @since x.x
+	 * @param {HTMLElement} option
+	 * @return {void}
+	 */
+	function deleteRelatedConditionalLogicOptions( option ) {
+		const deletedOptionValue = option.closest( '.frm_single_option' ).querySelector( '.frm_option_key input[type="text"]' ).value;
+		const rows               = builderPage.querySelectorAll( '.frm_logic_row' );
+
+		for ( let rowIndex = 0; rowIndex < rows.length; rowIndex++ ) {
```
Comment (truongwp) [inline js/formidable_admin.js:3537]: We can use `rows.forEach()` here.

File: js/formidable_admin.js [hunk 4]
```diff
@@ -4865,10 +4887,74 @@ function frmAdminBuildJS() {
...
+	 * @param {HTMLElement} choiceElement
+	 * @returns {Object}
+	 */
+	function getChoiceOldValueAndLabel( choiceElement ) {
+		const usingSeparateValues   = choiceElement.closest( '.frm-single-settings' ).querySelector( '.frm_toggle_sep_values' ).checked;
+		const singleOptionContainer = choiceElement.closest( '.frm_single_option' );
+
+		let oldValue, oldLabel;
+
+		// if ( choiceElement.parentElement.classList.contains( 'frm_single_option' ) ) { // label changed
```
Comment (truongwp) [inline js/formidable_admin.js:4938]: We should remove these commented code.

**Review by truongwp — APPROVED** [review]
This looks good to me. Thanks @AbdiTolesa!

**Review by Crabcyborg — COMMENTED** [review]
@AbdiTolesa Is it possible to move some of this code into the Pro `builder.js` file since this code doesn't need to be in Lite?

**Comment by AbdiTolesa** [comment]
> @AbdiTolesa Is it possible to move some of this code into the Pro `builder.js` file since this code doesn't need to be in Lite?

@Crabcyborg I have moved the function that deletes Conditional logic options to Pro in this PR: https://github.com/Strategy11/formidable-pro/pull/4940
The rest of the code seems to be tightly coupled with the current code.

**Comment by Crabcyborg** [comment]
@AbdiTolesa I think we'll need to update this PR now.

Your other PR (https://github.com/Strategy11/formidable-forms/pull/1894) also updates this code.

I just merged that other PR.

When you have the chance, can you work on fixing the merge conflict here?

Thank you!

**Comment by AbdiTolesa** [comment]
> @AbdiTolesa I think we'll need to update this PR now.
> 
> Your other PR (#1894) also updates this code.
> 
> I just merged that other PR.
> 
> When you have the chance, can you work on fixing the merge conflict here?
> 
> Thank you!

@Crabcyborg I have fixed the merge conflict now.

**Review by Crabcyborg — APPROVED** [review]
Thanks @AbdiTolesa @truongwp and @garretlaxton for working on this!

I think this is good to merge now.

**Pull request merged**